tt-train/sources/examples/nano_gpt/main.cpp

```diff
@@ -226,10 +230,22 @@ int main(int argc, char **argv) {
     fmt::print("Seed {}\n", ttml::autograd::ctx().get_seed());
     auto sequence_length = config.transformer_config.max_sequence_length;
 
-    auto [dataset, tokenizer] =
-        ttml::datasets::create_in_memory_token_dataset<ttml::tokenizers::CharTokenizer>(text, sequence_length);
+    auto create_dataset_and_tokenizer = [](const auto &text, const auto sequence_length, const auto &tokenizer_type) {
```

Comment:
mb use my approach with map like I did it in scheduler?

Reply:
Discussed offline. Keeping as is.